examples/chatgpt/src/chatgpt_bridge/actions/cutout.py

```diff
@@ -25,32 +25,47 @@ async def process(
     ctx: EditorAPIContext,
     stateid_input: StateID,
     background_color: str,
-    object_name: str,
+    prompt: str,
 ) -> tuple[Image.Image, StateID]:
-    # call infer-bbox
-    result_bbox = await ctx.call_async.infer_bbox(
+    # call detect
+    result_detect = await ctx.call_async.detect(
         state_id=stateid_input,
-        product_name=object_name,
+        prompt=prompt,
     )
-    if isinstance(result_bbox, ErrorResult):
-        raise ValueError(f"Cutout internal infer_bbox error: {result_bbox.error}")
-    bbox = result_bbox.bbox
-    app.logger.debug(f"{bbox=}")
-
-    # call segment
-    result_segment = await ctx.call_async.segment(
-        state_id=stateid_input,
-        bbox=bbox,
-    )
-    if isinstance(result_segment, ErrorResult):
-        raise ValueError(f"Cutout internal segment error: {result_segment.error}")
-    stateid_segment = result_segment.state_id
-    app.logger.debug(f"{stateid_segment=}")
+    if isinstance(result_detect, ErrorResult):
+        raise ValueError(f"Cutout internal detect error: {result_detect.error}")
+    detections = result_detect.results
+    app.logger.debug(f"{detections=}")
+
+    # call segment on each detection
+    stateids_segment = []
+    for detection in detections:
+        result_segment = await ctx.call_async.segment(
+            state_id=stateid_input,
+            bbox=detection.bbox,
+        )
+        if isinstance(result_segment, ErrorResult):
+            raise ValueError(f"Cutout internal segment error: {result_segment.error}")
+        stateids_segment.append(result_segment.state_id)
+    app.logger.debug(f"{stateids_segment=}")
+
+    # call merge-masks
```

Comment:
Well, it could happen that nothing got detected beforehand (results is empty). I think you want to stop the flow early, like here:

https://github.com/finegrain-ai/finegrain-python/blob/df07310daa078ec54b7ed31988b096142d14c3ea/examples/discord/src/finegrain_discord/bot.py#L160-L161